File: leetcode/1752_check_if_array_is_sorted_and_rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        count = 0
        n = len(nums)

        for i in range(len(nums)):
            if nums[i] > nums[(i + 1) % n]:
                count += 1
            if count > 1:
                return False

        return True

File: leetcode/1752_check_if_array_is_sorted_and_rotated.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function(nums) {
    let count = 0;
    let n = nums.length;

    for (let i = 0; i < n; i++) {
        // iterate through the array and check each adjacent pair
        // modulo for wrap around elements (checking first and last element)
        if (nums[i] > nums[(i + 1) % n]) {
            // if there is a break, where the current element is greater than the next element then add to our count
            count++;
        }
        // if there are multiple breaks, then it's impossible to sort with only one rotation
        if (count > 1) {
            return false;
        }
    }
    return true
};